File: Subprograms/ArrowFunction.js
// Arrow function
const square = (x) => x * x;

// Closure — inner function captures outer variable
function makeCounter(start = 0) {
  let count = start;
  return {
    increment: () => ++count,
    decrement: () => --count,
    value:     () => count
  };
}

const counter = makeCounter(10);
counter.increment();
counter.increment();
console.log(counter.value()); // 12

File: Subprograms/ArrowFunction.py
square = lambda x: x * x


class Counter:
    def __init__(self, start=0):
        self.count = start

    def increment(self):
        self.count += 1
        return self.count

    def decrement(self):
        self.count -= 1
        return self.count

    def value(self):
        return self.count


counter = Counter(10)
counter.increment()
counter.increment()
print(counter.value())
